### ISSUE TITLE
How much GPU memory is needed

### ISSUE THREAD

**Issue by ziyaxuanyi** [modelscope/Nexus-Gen#10]

**Comment by mi804** [comment]
The ar model`qwen2.5_vl` need about 20 GB, the vision decoder `Flux-1.dev` need about 35 GB. So 55 GB is needed if both model are loaded. But you may only need 35 GB for successfully usage if you offload `qwen2.5 vl` to cpu before you load `Flux-1.dev`

**Comment by ziyaxuanyi** [comment]
> The ar model`qwen2.5_vl` need about 20 GB, the vision decoder `Flux-1.dev` need about 35 GB. So 55 GB is needed if both model are loaded. But you may only need 35 GB for successfully usage if you offload `qwen2.5 vl` to cpu before you load `Flux-1.dev`

Thank you for your reply. By the way, is it still necessary to download the flux dev model separately?
The model address DiffSynth-Studio/Nexus-Gen seems to already contain the trained flux model: decoder_81_512.bin
And download_madels.py requires a separate download of the flux model: download_madels (["FLUX. 1-dev"])
Moreover, the inference code did not load the flux model ["FLUX.1-dev"] downloaded here separately, but instead loaded the decoder_81_512.bin

**Comment by mi804** [comment]
The download_models(["FLUX.1-dev"]) function downloads all model weights, including VAE, DiT, CLIP, and T5. While DiT is not actually used, the others are required. We will consider optimizing this part of the download and load code to remove the DiT portion.

**Comment by yang-0201** [comment]
> [@ziyaxuanyi](https://github.com/ziyaxuanyi)谢谢提醒。我们已经更新了下载和加载模型的代码，现在不再需要 FLUX 的原始 DiT 权重了。

Even with 48GB of VRAM, it still can't run now. Are there any solutions?

**Comment by mi804** [comment]
> > [@ziyaxuanyi](https://github.com/ziyaxuanyi)谢谢提醒。我们已经更新了下载和加载模型的代码，现在不再需要 FLUX 的原始 DiT 权重了。
> 
> Even with 48GB of VRAM, it still can't run now. Are there any solutions?

Please try [image_generation_offload.py](https://github.com/modelscope/Nexus-Gen/blob/main/image_generation_offload.py), which needs only about 24 GB VRAM

**Comment by yang-0201** [comment]
> > > [@ziyaxuanyi](https://github.com/ziyaxuanyi)谢谢提醒。我们已经更新了下载和加载模型的代码，现在不再需要 FLUX 的原始 DiT 权限了。
> > 
> > 
> > 即使有 48GB 的​​显存，现在仍然无法运行。有什么解决办法吗？
> 
> 请尝试[image_generation_offload.py](https://github.com/modelscope/Nexus-Gen/blob/main/image_generation_offload.py)，它只需要大约 24 GB 的 VRAM

Thank you vary much!

**Comment by yang-0201** [comment]
@mi804 One more question: Previously, when I fine-tuned Flux using the DiffSynth-Studio framework, 48GB of VRAM was sufficient. Now, when I use train_decoder.sh to fine-tune the visual encoder in Nexus-Gen, the VRAM is insufficient. Are there any optimization methods for this?

**Comment by mi804** [comment]
For full parameters finetuning of Flux DiT, 48GB of VRAM is insufficient. You may try to finetuning with lora, take [train_flux_lora](https://github.com/modelscope/DiffSynth-Studio/blob/main/examples/train/flux/train_flux_lora.py) for reference

**Comment by yang-0201** [comment]
> 对于 Flux DiT 的完整参数微调，48GB 的​​显存是不够的。您可以尝试使用 lora 进行微调，参考[train_flux_lora](https://github.com/modelscope/DiffSynth-Studio/blob/main/examples/train/flux/train_flux_lora.py)

Thank you for your reply, I understand. I took a look at the project structure. If we use LoRA fine-tuning, would we need to rewrite the training FluxForQwen file? Will you consider releasing LoRA fine-tuning code in the future? For general developers, full fine-tuning is quite costly.

**Comment by mi804** [comment]
Yes, if you want to perform LoRA fine-tuning, you should indeed modify the FluxForQwen class. In fact, within the Nexus-Gen architecture, the role of the Flux model is to perform image reconstruction based on image embeddings, making it less suitable for further fine-tuning. If you want to incorporate new task scenarios, it is recommended to fine-tune the LLM. However, if you do have a specific need for LoRA fine-tuning of Flux, we will consider adding support for it.

**Comment by yang-0201** [comment]
During my practical testing of Nexus-Gen's image editing capabilities, I observed that the model's generalization is somewhat limited. For instance, in specialized scenarios (not covered in the training dataset), it performs poorly—e.g., not only does the subject specified in the prompt change, but the surrounding context also undergoes significant alterations, resulting in a completely different output image. In contrast, Flux Fill with masking does not exhibit this issue.

To enhance the model's performance in this specific domain, I considered fine-tuning. However, full fine-tuning requires substantial computational resources and large datasets, which aren't feasible for our use case.

Do you have any recommendations for addressing this?

**Comment by mi804** [comment]
Regarding the issue of significant changes in the surrounding context, we have indeed been closely monitoring it. Similar problems exist in GPT-4o's image editing functionality, primarily due to two reasons: First, diffusion models are not strictly reconstruction models—they adjust their output based on their learned image distribution. Second, the process of encoding images into embeddings inevitably incurs information loss.

In our preliminary experiments, we found that increasing the number of Image Tokens input to Flux reduces information loss, improves reconstruction capability, and better preserves unedited regions. The current version of Nexus-Gen uses 81 Image Tokens, but we are considering training a 324-Token version, which is expected to significantly enhance the model's ability to maintain non-edited areas.

Additionally, the training data for Nexus-Gen is relatively limited compared to other text-to-image foundation models. As a result, it may be more sensitive to prompts during image generation and editing. We are considering increasing data diversity to mitigate this issue. We sincerely appreciate your attention to and support for this project.

**Comment by Jason-Chi-xx** [comment]
> mber of Image Tokens input to Flux reduces information loss, improves reconstruction capability, and better preserves unedited re

Hi, in your opinion, what length of image token would be suitable? In order to achieve a good editting quality. And I see the original qwen2.5-vl support arbitary image size, but you fix it. Is it for the better training and convergence? Can nexus-gen support arbitary image size.
